source/s3_meta_request.c

```diff
@@ -232,6 +234,54 @@ int aws_s3_meta_request_init_base(
     /* Keep original message around, for headers, method, and synchronous body-stream (if any) */
     meta_request->initial_request_message = aws_http_message_acquire(options->message);
 
+    if (options->receive_filepath.len > 0) {
+
+        meta_request->recv_filepath = aws_string_new_from_cursor(allocator, &options->receive_filepath);
+        switch (options->recv_file_options) {
+            case AWS_RECV_FILE_CREATE_OR_REPLACE:
+                meta_request->recv_file = aws_fopen(aws_string_c_str(meta_request->recv_filepath), "wb");
+                break;
+
+            case AWS_RECV_FILE_CREATE_NEW:
+                if (aws_path_exists(meta_request->recv_filepath)) {
+                    AWS_LOGF_ERROR(
+                        AWS_LS_S3_META_REQUEST,
+                        "id=%p The receive file already exists, but required to be not.",
+                        (void *)meta_request);
+                    aws_raise_error(AWS_ERROR_S3_RECV_FILE_EXISTS);
+                    break;
+                } else {
+                    meta_request->recv_file = aws_fopen(aws_string_c_str(meta_request->recv_filepath), "wb");
+                    break;
+                }
+            case AWS_RECV_FILE_CREATE_OR_APPEND:
+                meta_request->recv_file = aws_fopen(aws_string_c_str(meta_request->recv_filepath), "ab");
+                break;
+            case AWS_RECV_FILE_WRITE_TO_POSITION:
+                if (!aws_path_exists(meta_request->recv_filepath)) {
```

Comment:
may not need this check. you're calling `fopen()` with `r+` below, and docs say that gives an error if the file doesn't exist https://en.cppreference.com/w/c/io/fopen

Reply:
Similar to the previous one, without the extra check, we will raise `AWS_ERROR_FILE_INVALID_PATH`, which cannot tell the issue is really from the configuration.